app/scripts/views/legal_copy.js

```diff
@@ -72,12 +45,25 @@ define(function (require, exports, module) {
         // it on other requests, and to ensure the server
         // only sends the user back to the app if the user in fact
         // came from this page.
-        self.window.document.cookie = 'canGoBack=1; path=' + self.window.location.pathname;
+        this.window.document.cookie = 'canGoBack=1; path=' + this.window.location.pathname;
+        return proto.afterRender.call(this);
       })
-      .fail(function () {
-        self.displayError(self.fetchError);
-        self.$('.hidden').removeClass('hidden');
+      .fail(() => {
+        this.displayError(this.fetchError);
+        this.$('.hidden').removeClass('hidden');
       });
+    },
+
+    afterVisible () {
+      this.$('.hidden').removeClass('hidden');
+
+      // The `data-shown` attribute is searched for by the functional
+      // tests. The legal copy HTML has unstable markup, so the functional
+      // tests need some stable identifier to look for to know the copy is
+      // ready.
+      this.$('#legal-copy').attr('data-shown', 'true');
```

Comment:
Since functional tests look for this attribute, I thought we should only add it once the template is actually visible.